### PR TITLE
add workflow for branch deploys

### PR DESCRIPTION
**Pull request by ssube**

### CODE REVIEW

File: .github/workflows/deploy-branch.yml
Comment (lrojas94) [inline .github/workflows/deploy-branch.yml:5]: Nit - I think it makes sense to publish with new commits too.  If changes are requested, I doubt we wanna wait for another review to deploy them

**Review by lrojas94 — APPROVED** [review]

File: .github/workflows/deploy-branch.yml
Comment (ssube) [reply .github/workflows/deploy-branch.yml:5]: I looked at that event as well, but we need to make sure that only core developers can trigger a release, so I had to base the workflow on something only core developers can do. Approval seemed like the best option, but I'm open to ideas.

Comment (ssube) [reply .github/workflows/deploy-branch.yml:9]: I tested a few different methods, including `pull_request.author_association`, but it looks like the best way to make sure only dev can deploy will be to limit actions to this repo (turn them off for forks) and then apply branch rules as needed.

**Review by manuelurenah — APPROVED** [review]

**Review by lrojas94 — APPROVED** [review]

**Pull request merged**
